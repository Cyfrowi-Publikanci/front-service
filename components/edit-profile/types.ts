export interface EditFormObject {
  email: string;
  password: string;
  passwordConfirm: string;
}

export type HandleFormSubmitType = (values: EditFormObject) => void;
