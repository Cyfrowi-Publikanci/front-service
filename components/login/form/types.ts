export interface LoginFormObject {
  email: string;
  password: string;
}

export type HandleFormSubmitType = (values: LoginFormObject) => void;
