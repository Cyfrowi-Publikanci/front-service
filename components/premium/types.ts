export interface BuyPremiumFormObject {
    card: string;
    cvc: string;
    month: string;
    year: string;
  }
  
  export type HandleFormSubmitType = (values: BuyPremiumFormObject) => void;