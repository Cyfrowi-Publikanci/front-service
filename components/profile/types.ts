export interface ProfileFormObject {
  id: string;
  preferences: string;
}

export type HandleFormSubmitType = (values: ProfileFormObject) => void;
