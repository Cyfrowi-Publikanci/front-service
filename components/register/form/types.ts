export interface RegisterFormObject {
  email: string;
  password: string;
}

export type HandleFormSubmitType = (values: RegisterFormObject) => void;
