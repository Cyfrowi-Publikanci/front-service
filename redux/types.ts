import { AlertType } from '../components/common/alerts/types';

export interface AuthState {
  authorization: string;
  authenticateError: boolean;
  state: string,
}

export interface ComponentsState {
  loader: boolean;
  showAlert: boolean;
  alert: {
    message: string;
    type?: AlertType;
  };
}

export interface ProfileState {
  bgColor: string;
  fontSize: string,
}
