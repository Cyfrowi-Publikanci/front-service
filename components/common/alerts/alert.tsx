import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Snackbar } from '@material-ui/core';
import Alert from '@material-ui/lab/Alert/Alert';

import { selectAlert } from '../../../redux/selectors';
import { hideAlert } from '../../../redux/actions/components';

export function AlertPubLab() {
  const dispatch = useDispatch();
  const alert = useSelector(selectAlert);

  const onClose = useCallback(() => {
    dispatch(hideAlert());
  }, []);

  if (!alert.message) return <></>;

  return (
    <Snackbar
      open={alert.message.length > 0}
      anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
      autoHideDuration={4000}
      onClose={onClose}
    >
      <Alert variant="filled" onClose={onClose} severity={alert.type}>
        {alert.message}
      </Alert>
    </Snackbar>
  );
}
