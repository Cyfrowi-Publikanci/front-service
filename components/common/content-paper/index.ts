export { ContentPaper } from './content-paper';
