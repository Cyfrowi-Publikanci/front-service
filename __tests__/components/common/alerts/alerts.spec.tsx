import { shallow } from 'enzyme';
import * as reactRedux from 'react-redux';

import { AlertPubLab } from '../../../../components/common/alerts';
import { AlertType } from '../../../../components/common/alerts/types';

jest.mock('react-i18next', () => ({ useTranslation: () => ({ t: (key: string) => key }) }));

jest.mock('react-redux');

describe('AlertPubLab', () => {
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');

  beforeEach(() => {
    useSelectorMock.mockReset();
  });

  it('should match snapshot if alert type is SUCCESS', () => {
    useSelectorMock.mockReturnValueOnce({
      message: 'Test',
      type: AlertType.SUCCESS,
    });
    const wrapper = shallow(<AlertPubLab />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if alert type is ERROR', () => {
    useSelectorMock.mockReturnValueOnce({
      message: 'Test',
      type: AlertType.ERROR,
    });
    const wrapper = shallow(<AlertPubLab />);

    expect(wrapper).toMatchSnapshot();
  });
});
