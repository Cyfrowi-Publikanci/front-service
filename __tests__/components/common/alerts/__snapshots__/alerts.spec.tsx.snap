// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPubLab should match snapshot if alert type is ERROR 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "bottom",
    }
  }
  autoHideDuration={4000}
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(Alert))
    onClose={[Function]}
    severity="error"
    variant="filled"
  >
    Test
  </WithStyles(ForwardRef(Alert))>
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`AlertPubLab should match snapshot if alert type is SUCCESS 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "bottom",
    }
  }
  autoHideDuration={4000}
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(Alert))
    onClose={[Function]}
    severity="success"
    variant="filled"
  >
    Test
  </WithStyles(ForwardRef(Alert))>
</WithStyles(ForwardRef(Snackbar))>
`;
